PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -427,8 +427,12 @@ LoadAndSetupImage (
     }
     if (FeaturePcdGet (PcdMeasuredBootEnabled) && (LoaderPlatformInfo->LdrFeatures & FEATURE_MEASURED_BOOT)) {
       // Extend hash of the image into TPM.
-      TpmExtendPcrAndLogEvent ( 8, TPM_ALG_SHA256, LoadedImage->ImageHash,
-        EV_COMPACT_HASH, sizeof("LinuxLoaderPkg: OS Image"), (UINT8 *)"LinuxLoaderPkg: OS Image");
+      TpmExtendPcrAndLogEvent ( 8, 
+        (TPMI_ALG_HASH) GetTpmHashAlgEnabled(),
```

Comment:
This is not correct. You need to pass the hash algorithm corresponding to the hash(3rd arg). TpmExtendPcrAndLogEvent() API would extend that hash if the corresponding PCR bank is enabled, else it will throw a error.